synapse/state/__init__.py

```diff
@@ -420,6 +420,69 @@ async def compute_event_context(
             partial_state=partial_state,
         )
 
+    async def compute_event_context_for_batched(
+        self,
+        event: EventBase,
+        state_ids_before_event: StateMap[str],
+        current_state_group: int,
+    ) -> EventContext:
```

Comment:
Is this now basically the same as `state_ids_before_event`, except with an extra `current_state_group`?

Reply:
Actually, I guess we'll want to keep these functions different as we'll be changing this in future iterations?